samples/bluetooth/periodic_sync/src/main.c

```diff
@@ -78,6 +78,10 @@ static void scan_recv(const struct bt_le_scan_recv_info *info,
 
 	bt_data_parse(buf, data_cb, name);
 
+	if (strcmp(name, "Test Periodic Advertising")) {
+		return;
+	}
```

Comment:
That seems a bit arbitrary. Where did you take this specific name from? I guess if you want to be able to configure this app to filter for a specific remote name, then you could add an app-specific Kconfig option for that.

Reply:
It is a Kconfig option of the advertising sample: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/periodic_adv/prj.conf
So yeah, probably need to find a way to share common configuration between the two samples, since there's no reason why a user may not have customized the message sent by the advertising sample.

Reply:
Keep in mind that these sample do not need to be run with their "counterparts", i.e. this sample can be run without running the `periodic_adv` sample on another board, and thus should not have any dependencies. 

I'm not against adding a name check, but it needs to be optional. See e.g. how we do this in the Broadcast Audio Sink sample: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/broadcast_audio_sink/Kconfig#L31-L36

Reply:
I just test `periodic_adv` and `periodic_sync` demo but it dosn't work due to other advertising. So I upstream this PR.
I'll add into Kconfig option for that.